src/Controller/Cliente/ClienteFormController.php

```diff
@@ -20,31 +21,53 @@ public function __construct(ClienteService $clienteService)
     }
 
     #[Route("/clientes/adicionar", name: "RegistrarCliente", methods: ['POST'])]
-    public function registrarCliente(Request $request,ValidarCpfService $validarCpfService): Response
+    public function registrarCliente(Request $request,ValidarCpfService $validarCpfService, ClienteRepository $clienteRepository): Response
     {
         $token = $request->request->get("_csrf_token");
 
         if (!$this->isCsrfTokenValid("addCliente", $token)) {
             $this->addFlash("danger", "Token CRSF inválido!");
             return $this->redirectToRoute("clientes");
         }
+
+        if(!$validarCpfService->execute($request->request->get("cpf"))){
+            $this->addFlash("danger", "CPF inválido!");
+            return $this->redirectToRoute("adicionarCliente");
+        }
         
         $cliente = new Cliente();
 
-        if(!$validarCpfService->execute($request->request->get("cpf"))){
-            $this->addFlash("danger", "CPF invalido inválido!");
-            return $this->redirectToRoute("clientes");
+        $cpf = $request->request->get("cpf");
+        $cpf = str_replace(['.', '-'], '', $cpf);
+
+        if($clienteRepository->buscarClienteCPF($cpf)){
+            $this->addFlash("danger", "CPF já cadastrado!");
+            return $this->redirectToRoute("adicionarCliente");
         }
 
-        $cliente->setCpf((int) $request->request->get("cpf"));
-        $cliente->setNome((string) $request->request->get("nome"));
+        $cliente->setCpf((int) $cpf);
+
+        if(is_numeric($request->request->get("nome"))){
+            $this->addFlash("danger", "O nome inserido contém apenas números!");
+            return $this->redirectToRoute("adicionarCliente"); 
+        }
+        $nome = trim((string) $request->request->get("nome"));
+        $nome = preg_replace('/\s+/', ' ', $nome); 
+
+        if(empty($nome)){
+            $this->addFlash("danger", "O nome inserido está vazio!");
+            return $this->redirectToRoute("adicionarCliente"); 
+        }
+
+        $cliente->setNome((string) $nome);
```

Comment:
passar no construtor também, ai aproveita e faz um new Cliente depois de todas as validações
```suggestion
new Cliente($nome, $cpf)
```